### PR TITLE
feat(module:table): support nzFixed for nzSummary

### PR DESCRIPTION
**Pull request by Laffery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

see also https://ant-design.antgroup.com/components/table#components-table-demo-summary

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: components/table/doc/index.en-US.md [hunk 1]
```diff
@@ -23,37 +23,42 @@ The Table component is both easy to use and highly customizable.
...
+  <tr *ngFor="let data of basicTable.data">
+    <td>{{data.name}}</td>
+    <td>{{data.age}}</td>
+    <td>{{data.address}}</td>
+    <td>
+      <a>Action 一 {{data.name}}</a>
+      <nz-divider nzType="vertical"></nz-divider>
+      <a>Delete</a>
+    </td>
+  </tr>
```
Comment (HyperLife1119) [inline components/table/doc/index.en-US.md:61]: 这是预期的缩进格式变更吗？

Comment (Laffery) [reply components/table/doc/index.en-US.md:61]: ide格式化markdown代码块有点问题，我改下

Comment (Laffery) [reply components/table/doc/index.en-US.md]: done

File: components/table/src/table/tfoot-summary.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  ViewEncapsulation
+} from '@angular/core';
+
+import { NzSafeAny } from 'ng-zorro-antd/core/types';
+
+import { NzTableStyleService } from '../table-style.service';
+import { NzTableSummaryFixedType } from '../table.types';
+
+function fixedAttribute(value: NzTableSummaryFixedType | boolean | unknown): NzTableSummaryFixedType | null {
+  return value === 'top' || value === 'bottom' ? value : booleanAttribute(value) ? 'bottom' : null;
```
Comment (HyperLife1119) [inline components/table/src/table/tfoot-summary.component.ts:27]:
```suggestion
  return typeof value === 'string' ? value : booleanAttribute(value) ? 'bottom' : null;
```

File: components/table/src/table/tfoot-summary.component.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+  `,
+  imports: [NgTemplateOutlet],
+  host: {
+    '[class.ant-table-summary]': '!isInsideTable || !nzFixed'
+  }
+})
+export class NzTfootSummaryComponent implements OnInit, OnChanges {
+  @Input({ transform: fixedAttribute }) nzFixed: NzTableSummaryFixedType | null = null;
+  isInsideTable = false;
```
Comment (HyperLife1119) [inline components/table/src/table/tfoot-summary.component.ts:51]:
```suggestion
  isInsideTable = !!this.nzTableStyleService;
```

File: components/table/src/table/tfoot-summary.component.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  @ViewChild('contentTemplate', { static: true }) templateRef!: TemplateRef<NzSafeAny>;
+  private nzTableStyleService = inject(NzTableStyleService, { optional: true });
+
+  ngOnInit(): void {
+    if (this.nzTableStyleService) {
+      this.isInsideTable = true;
+      this.nzTableStyleService.setTfootTemplate(this.templateRef);
+    }
```
Comment (HyperLife1119) [inline components/table/src/table/tfoot-summary.component.ts:60]:
```suggestion
    this.nzTableStyleService?.setTfootTemplate(this.templateRef);
```

File: components/table/src/table/tfoot-summary.component.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    const { nzFixed } = changes;
+    if (this.nzTableStyleService) {
+      this.nzTableStyleService.setTfootFixed(nzFixed.currentValue);
+    }
```
Comment (HyperLife1119) [inline components/table/src/table/tfoot-summary.component.ts:67]:
```suggestion
    this.nzTableStyleService?.setTfootFixed(nzFixed.currentValue);
```

File: components/table/demo/summary.ts [hunk 6]
```diff
@@ -37,15 +38,38 @@ import { Component, OnInit } from '@angular/core';
...
     </nz-table>
-  `,
-  styles: [
-    `
-      tfoot th,
-      tfoot td {
-        background: #fafafa;
-      }
-    `
-  ]
```
Comment (HyperLife1119) [inline components/table/demo/summary.ts:48]: 我认为可以保留这些 demo 样式，WDYT？

Comment (Laffery) [reply components/table/demo/summary.ts]: agree with you

File: components/table/src/table/tfoot-summary.component.ts (hunk 2 above)
Comment (Laffery) [reply components/table/src/table/tfoot-summary.component.ts:27]: 如果 value 为 `string` 类型且值不为字面量 `top` `bottom` ，则 transform 结果为 value 字符串本身。
而在 nz-table-inner-scroll 中是通过 `nzFixed` 的值来判断是固定在顶部还是在底部，由于其值不为 `top` `bottom` 任一个，最终 tfootTemplate 不会在 nz-table-inner-scroll 中渲染

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Comment by OriginRing** [comment]
添加 nzLeft 背景色丢失
截屏2024-07-23 18 17 08

**Comment by OriginRing** [comment]
对于添加 tfoot 后 tbody 横向滚动条能不能隐藏掉🤔（不是很好看）

**Comment by Laffery** [comment]
> ant-design.antgroup.com/components/table

样式同 antd

image

**Comment by Laffery** [comment]
> 添加 nzLeft 背景色丢失

nzLeft 引入了样式覆盖了 tfoot 中的背景色。

```css
.ant-table-cell-fix-left, .ant-table-cell-fix-right {
  position: sticky !important;
  z-index: 2;
  background: #fff;
}
```

这里 tfoot 的背景色为 demo 中特别设置的样式，而不是组件本身的样式，所以我认为不影响

**Comment by OriginRing** [comment]
LGTM

**Review by OriginRing — APPROVED** [review]

**Pull request merged**